moveit_kinematics/kdl_kinematics_plugin/include/moveit/kdl_kinematics_plugin/kdl_kinematics_plugin.h

```diff
@@ -122,84 +120,62 @@ class KDLKinematicsPlugin : public kinematics::KinematicsBase
 
-  /** @brief Check whether the solution lies within the consistency limit of the seed state
+  /** @brief Check whether the solution lies within the consistency limits of the seed state
    *  @param seed_state Seed state
-   *  @param redundancy Index of the redundant joint within the chain
-   *  @param consistency_limit The returned state for redundant joint should be in the range
-   * [seed_state(redundancy_limit)-consistency_limit,seed_state(redundancy_limit)+consistency_limit]
+   *  @param consistency_limits
    *  @param solution solution configuration
    *  @return true if check succeeds
    */
-  bool checkConsistency(const KDL::JntArray& seed_state, const std::vector<double>& consistency_limit,
-                        const KDL::JntArray& solution) const;
-
-  int getJointIndex(const std::string& name) const;
+  bool checkConsistency(const Eigen::VectorXd& seed_state, const std::vector<double>& consistency_limits,
+                        const Eigen::VectorXd& solution) const;
 
-  int getKDLSegmentIndex(const std::string& name) const;
+  void getRandomConfiguration(Eigen::VectorXd& jnt_array) const;
 
-  void getRandomConfiguration(KDL::JntArray& jnt_array, bool lock_redundancy) const;
-
-  /** @brief Get a random configuration within joint limits close to the seed state
+  /** @brief Get a random configuration within consistency limits close to the seed state
    *  @param seed_state Seed state
-   *  @param redundancy Index of the redundant joint within the chain
-   *  @param consistency_limit The returned state will contain a value for the redundant joint in the range
-   * [seed_state(redundancy_limit)-consistency_limit,seed_state(redundancy_limit)+consistency_limit]
+   *  @param consistency_limits
    *  @param jnt_array Returned random configuration
    */
-  void getRandomConfiguration(const KDL::JntArray& seed_state, const std::vector<double>& consistency_limits,
-                              KDL::JntArray& jnt_array, bool lock_redundancy) const;
-
-  bool isRedundantJoint(unsigned int index) const;
+  void getRandomConfiguration(const Eigen::VectorXd& seed_state, const std::vector<double>& consistency_limits,
+                              Eigen::VectorXd& jnt_array) const;
 
-  bool active_; /** Internal variable that indicates whether solvers are configured and ready */
-
-  moveit_msgs::KinematicSolverInfo ik_chain_info_; /** Stores information for the inverse kinematics solver */
-
-  moveit_msgs::KinematicSolverInfo fk_chain_info_; /** Store information for the forward kinematics solver */
-
-  KDL::Chain kdl_chain_;
+  bool initialized_;  //< Internal variable that indicates whether solver is configured and ready
 
-  unsigned int dimension_; /** Dimension of the group */
-
-  KDL::JntArray joint_min_, joint_max_; /** Joint limits */
-
-  mutable random_numbers::RandomNumberGenerator random_number_generator_;
+  unsigned int dimension_;                        //< Dimension of the group
```

Comment:
nit: why the "<" for comments? seems non-standard

Reply:
http://www.doxygen.nl/manual/docblocks.html#memberdoc
But, indeed I missed the third slash...